src/lelantus.cpp

```diff
@@ -1189,59 +1212,11 @@ bool CLelantusState::HasCoinHash(GroupElement &pubCoinValue, const uint256 &pubC
     return false;
 }
 
-bool CLelantusState::HasReducedCoinHash(GroupElement &pubCoinValue, const uint256 &pubCoinValueHash) {
-    auto const& mints = GetMints();
-    auto& reducedHashToGroupElement = containers.GetReducedHashToGroupElement();
-    if(pwalletMain->IsLocked()) {
-        return false;
-    }
-    for ( auto it = mints.begin(); it != mints.end(); ++it ){
-        const lelantus::PublicCoin & pubCoin = (*it).first;
-        COutPoint outPoint;
-        GetOutPoint(outPoint, pubCoin);
-        uint256 hashBlock;
-        CTransactionRef tx;
-        if (!GetTransaction(outPoint.hash, tx, ::Params().GetConsensus(), hashBlock, true)) {
-            continue;
-        }
-        for (const CTxOut& out : tx->vout) {
-            if (!out.scriptPubKey.IsLelantusMint() && !out.scriptPubKey.IsLelantusJMint()) {
-                continue;
-            }
-            secp_primitives::GroupElement pubcoin;
-            lelantus::ParseLelantusMintScript(out.scriptPubKey, pubcoin);
-            bool skip = false;
-            for (auto it = reducedHashToGroupElement.begin(); it != reducedHashToGroupElement.end(); ++it) {
-                if (it->second == pubcoin)
-                    skip = true;
-            }
-            if(skip)
-                continue;
-
-            uint64_t amount  = 0;
-            try {
-                if (out.scriptPubKey.IsLelantusMint()) {
-                    amount = out.nValue;
-                }  else {
-                    std::vector<unsigned char> encryptedValue;
-                    lelantus::ParseLelantusJMintScript(out.scriptPubKey, pubcoin, encryptedValue);
-                    if(!pwalletMain->DecryptMintAmount(encryptedValue, pubcoin, amount))
-                        continue;
-                }
-            } catch (std::invalid_argument&) {
-                continue;
-            }
-
-            uint256 hashPubcoin = primitives::GetPubCoinValueHash(pubcoin + lelantus::Params::get_default()->get_h1() * Scalar(amount).negate());
-            reducedHashToGroupElement[hashPubcoin] = pubcoin;
-        }
-    }
-
-    for (auto it = reducedHashToGroupElement.begin(); it != reducedHashToGroupElement.end(); ++it ) {
-        if ((*it).first == pubCoinValueHash) {
-            pubCoinValue = (*it).second;
-            return true;
-        }
+bool CLelantusState::HasReducedCoinHash(GroupElement& pubCoinValue, const uint256& pubCoinValueHash) {
```

Comment:
Again should choose one name and stick to it. It's not `pubCoinValueHash` that's passed to this function. And it's not reduced